Sources/Helpers/OASvgHelper.mm

```diff
@@ -10,11 +10,21 @@
 #import "OANativeUtilities.h"
 
 #include <OsmAndCore/SkiaUtilities.h>
+#include <QReadWriteLock>
 
 const static float kDefaultIconSize = 24.0f;
 
+static NSMutableDictionary<NSString *, NSString *> *resourcesExistCache = [NSMutableDictionary dictionary];
```

Comment:
It is better to have single cache `resourcesPaths` with empty string if resource does not exist.